src/test/java/io/vertx/test/core/HttpTest.java

```diff
@@ -2660,6 +2671,57 @@ private void testPooling(boolean keepAlive, boolean pipelining) {
   }
 
   @Test
+  public void testMaxWaitQueueSizeIsRespected() throws Exception {
+    client.close();
+
+    client = vertx.createHttpClient(new HttpClientOptions().setDefaultHost(DEFAULT_HTTP_HOST).setDefaultPort(DEFAULT_HTTP_PORT)
+        .setPipelining(false).setMaxWaitQueueSize(0).setMaxPoolSize(2));
+
+    AtomicBoolean tooBusyExceptionOccurred = new AtomicBoolean(false);
+    CountDownLatch latch = new CountDownLatch(3);
+
+    server.requestHandler(req -> {
+      req.response().setStatusCode(200);
+      req.response().end("OK");
+    });
+
+    server.listen(onSuccess(s -> {
+      HttpClientRequest req1 = client.get(DEFAULT_TEST_URI, resp -> {
+        resp.bodyHandler(body -> {
+          assertEquals("OK", body.toString());
+          latch.countDown();
+        });
+      });
+      req1.exceptionHandler(t -> fail("Should not be called."));
+
+      HttpClientRequest req2 = client.get(DEFAULT_TEST_URI, resp -> {
+        resp.bodyHandler(body -> {
+          assertEquals("OK", body.toString());
+          latch.countDown();
+        });
+      });
+      req2.exceptionHandler(t -> fail("Should not be called."));
+
+      HttpClientRequest req3 = client.get(DEFAULT_TEST_URI, resp -> {
+        fail("Should not be called.");
+      });
+      req3.exceptionHandler(t -> {
+        assertTrue("Incorrect exception time.", t instanceof ConnectionPoolTooBusyException);
+        tooBusyExceptionOccurred.set(true);
+        latch.countDown();
```

Comment:
I can try that but technically the second request marks the end of the test because we expect the third to receive an exception immediately upon calling end() since the pool size is 2 and maxWaitQueueSize is 0.